kolena/detection/_internal/test_case.py

```diff
@@ -125,6 +127,7 @@ def _from(cls, response: CoreAPI.EntityData) -> "BaseTestCase":
         return obj
 
     @classmethod
+    @with_event(event_name=EventAPI.Event.CREATE_TEST_CASE)
```

Comment:
I am new to this, would we be able to distinguish between CREATE_TEST_CASE Event for detection and fr?

Reply:
Not at the moment. We can expand this decorator to take additional args as a next step if the team want, but currently my understanding is that we are primarily interested in usage metrics.

Reply:
Could be beneficial when analyzing the usage of these legacy stuff, but I am ok to do it after we have the need to have the requirement